### PR TITLE
`General`: Add time zone information to date time picker

### PR DESCRIPTION
**Pull request by filip-gregurevic**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [ ] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-testing/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

While setting the release date for a course/exercise/lecture you are unable to see the timezone of the date you are setting. It seems like that Artemis is always taking the local timezone of the user. If you are for example in an other timezone than your students, you might get offsets for the dates.

For example if you are in EEST (UTC +3) and set the release time for 2021-09-01T08:00:00 the date which will be shown to students in Munich (CEST (UTC +2)) is 2021-09-01T07:00:00.

Closes https://github.com/ls1intum/Artemis/issues/3966

### Description

Added an optional time zone warning to the date picker component and used it in course, lecture and exercise creation.

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Navigate to Course Administration
3. Create a course
4. Check that the time zone warning is correct and displayed
5. Switch time zone on your computer
6. Check the time zone again
7. Repeat for lecture and exercises

#### Code Review
- [x] Review 1
- [ ] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### Screenshots

Bildschirmfoto 2021-10-24 um 14 18 22
Bildschirmfoto 2021-10-24 um 14 18 16

### CODE REVIEW

File: src/main/webapp/app/shared/date-time-picker/date-time-picker.component.ts [hunk 1]
```diff
@@ -44,6 +51,7 @@ export class FormDateTimePickerComponent implements ControlValueAccessor {
     @Input() startAt: dayjs.Dayjs = dayjs().startOf('minutes'); // Default selected date. By default this sets it to the current time without seconds or milliseconds;
     @Input() min: dayjs.Dayjs; // Dates before this date are not selectable.
     @Input() max: dayjs.Dayjs; // Dates after this date are not selectable.
+    @Input() timeZoneWarning: boolean; // Displays a warning that the current time zone might differ from the participants'.
```
Comment (Olimbia23) [inline src/main/webapp/app/shared/date-time-picker/date-time-picker.component.ts:54]: [nit] maybe rename it to `shouldShowTimeZoneWarning`?

Comment (Olimbia23) [inline src/main/webapp/app/shared/date-time-picker/date-time-picker.component.ts:54]: [nit] it seems like the value is set to `true` everywhere. What do you think of setting this value to `true` by default and to not explicitly set it in html?

**Review by Olimbia23 — APPROVED** [review]
Good work @filip-gregurevic.
Code changes look fine. I left two minor comments.
Approve 👍

File: src/main/webapp/app/shared/date-time-picker/date-time-picker.component.ts (hunk 1 above)
Comment (filip-gregurevic) [reply src/main/webapp/app/shared/date-time-picker/date-time-picker.component.ts]: Thank you for the review and the feedback, I implemented it.

**Comment by krusche** [comment]
Looks good now. Thank you!

**Pull request merged**